packages/rum/src/domain/record/observer.ts

```diff
@@ -399,3 +424,23 @@ function initVisualViewportResizeObserver(cb: VisualViewportResizeCallback): Lis
     cancelThrottle()
   }
 }
+
+function initFrustrationObserver(lifeCycle: LifeCycle, frustrationCb: FrustrationCallback): ListenerHandler {
+  return lifeCycle.subscribe(LifeCycleEventType.RAW_RUM_EVENT_COLLECTED, (data) => {
+    if (
+      data.rawRumEvent.type === RumEventType.ACTION &&
+      data.rawRumEvent.action.type === ActionType.CLICK &&
+      data.rawRumEvent.action.frustration?.type &&
+      'event' in data.domainContext &&
+      data.domainContext.event
+    ) {
+      const frustrationType = getFrustrationFromAction(data.rawRumEvent.action.frustration.type)
+      frustrationCb({
+        timestamp: data.rawRumEvent.date,
+        frustrationType,
+        recordIds:
+          frustrationType === FrustrationType.RAGE_CLICK ? [] : [getRecordIdForEvent(data.domainContext.event)],
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏even if we don't support all events of RAGE_CLICK yet, you could remove this special case and report the first event/record of rage clicks.